opencog/persist/load_scm/fast_load.cc

```diff
@@ -89,10 +89,7 @@ static std::string get_next_token(const std::string& s, uint& l, uint& r)
         uint l1 = l;
         for(; l1 < r && (s[l1] != '"' or ((0 < l1) and (s[l1 - 1] == '\\'))); l1++)
         {
-            // Unescape esaped quotes.
-            // XXX is there anything else we want to unescape?
-            if (s[l1] != '\\' or s[l1+1] != '"')
-                token.push_back(s[l1]);
+            token.push_back(s[l1]);
```

Comment:
!? But this is gong to result with a `\"` in the atom name ... there is nothing in the atomspace that will string this out later, and so you will now get an atom with a broken name in it...

Reply:
(ConceptNode "test\"") produces valid atom. In what way it is broken?

Reply:
I don't understand ... if you pass `(ConceptNode "test"")` into your tokenizer, it will complain when it finds the third quote-mark, right? Because the second quote ends the token search, and the third quote is going to mess up the close-paren... maybe I am reading your code incorrectly?

Reply:
it fails on cases with backslash before the last token: like that 
 ```"test\\"```

I need to add backlash counting or something to properly handle such cases, currently condition just checks that the symbol previous to " is backlash, which is not correct.

While "test\"" is ok.

Reply:
@linas i mean ```(ConceptNode "test\"") ```, not ```(ConceptNode "test"") ```  - which is not valid of course

Reply:
Yes, OK, I understand. In general, unescaping backslashes during file-load is generically necessary. But this is your code, in the end you need to maintain it and get it working correctly.  I tried and failed :-/